trecs/components/users.py

```diff
@@ -361,6 +378,11 @@ def get_user_feedback(self, *args, **kwargs):
             self.update_profiles(interact_attrs)
             # update user scores
             self.compute_user_scores(item_attributes)
+        # record interactions if needed to ensure users don't repeat interactions
+        if not self.repeat_interactions:
```

Comment:
Is there a reason why we need this conditional at both line 354 and here? If not, could we combine the contents for parsimony?

Reply:
yeah, the conditionals are doing slightly different things - the first one is altering the scores matrix so the recommendations *sent* to users exclude the items they've already interacted with, the second is *taking in* the user interactions (after they've chosen their desired item from the recommendation set), so they kind of have to be in these two different places. let me know if that made sense!